src/OpenTelemetry/Trace/TracerProviderBuilder.cs

```diff
@@ -130,6 +158,108 @@ public TracerProviderBuilder AddActivitySource(string activitySourceName)
             return this;
         }
 
+        public TracerProvider Build()
+        {
+            this.Sampler = this.Sampler ?? new ParentOrElseSampler(new AlwaysOnSampler());
+
+            var provider = new TracerProviderSdk
+            {
+                Resource = this.Resource,
+                Sampler = this.Sampler,
+                ActivityProcessor = this.ActivityProcessor,
+            };
+
+            var activitySource = new ActivitySourceAdapter(provider.Sampler, provider.ActivityProcessor, provider.Resource);
```

Comment:
activitysourceadapter won't fech ActiivtyProcessors added after building the provider. its a TODO on my list.